module/mfrc522_spi.c

```diff
@@ -11,3 +11,24 @@ struct address_byte address_byte_build(u8 mode, u8 addr)
 
 	return byte;
 }
+
+u8 mfrc522_register_read(struct spi_device *client, u8 reg)
+{
+	struct address_byte reg_read =
+		address_byte_build(MFRC522_SPI_READ, reg);
+	u8 value;
+
+	spi_write(client, &reg_read, 1);
+	spi_read(client, &value, 1);
+
+	return value;
+}
+
+void mfrc522_register_write(struct spi_device *client, u8 reg, u8 value)
+{
+	struct address_byte reg_write =
+		address_byte_build(MFRC522_SPI_WRITE, reg);
+
+	spi_write(client, &reg_write, 1);
+	spi_write(client, &value, 1);
```

Comment:
Maybe we could create an array and only do one call to `spi_write`?
```suggestion
#define REG_WRITE_LEN 2
        u8 tx_buffer[REG_WRITE_LEN] = { reg_write, value };
	spi_write(client, tx_buffer, REG_WRITE_LEN);
#undef REG_WRITE_LEN
```
This way we only need to acquire the various SPI ressources such as locks once

Reply:
Good idea